### PR TITLE
docs: initial overview docs for button - dialog

### PR DESCRIPTION
**Pull request by jelbourn**
Initial overview docs for button, button-toggle, card, checkbox, and dialog. Will need more examples added later.

cc @topherfangio @DevVersion if you want to do something similar for your components

### CODE REVIEW

File: src/lib/dialog/OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+### Dialog content
+Several directives are available to make it easier to structure your dialog content:
+
+| Name                  | Description                                                              |
+|-----------------------|--------------------------------------------------------------------------|
+| `md-dialog-title`     | \[Attr] Dialog title, applied to a heading element (e.g., `<h1>`, `<h2>`)|
+| `<md-dialog-content>` | Primary scrollable content of the dialog                                 |
+| `<md-dialog-actions>` | Container for action buttons at the bottom of the dialog                 |
+| `md-dialog-close`     | \[Attr] Added to a `<button>` to make the button close the on click      |
```
Comment (crisbeto) [inline src/lib/dialog/OVERVIEW.md]: You missed "dialog" here - "...to make the button close the dialog on click".

**Comment by DevVersion** [comment]
@jelbourn Yeah I will do the docs for the slide-toggle and toolbar.

File: src/lib/button-toggle/OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Buttons toggles aregroups of buttons that can be toggled on and off. They can be configured to
```
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: Button toggles are groups...

File: src/lib/button-toggle/OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Buttons toggles aregroups of buttons that can be toggled on and off. They can be configured to 
+behave as checkboxes or as radio buttons. Button toggles are always part of a 
+`md-button-toggle-group`.
```
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: I don't think this is true, you could have a single button that can toggle on or off. The code says the toggle group is optional: https://github.com/angular/material2/blob/master/src/lib/button-toggle/button-toggle.ts#L299

File: src/lib/button-toggle/OVERVIEW.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+<!-- example(button-toggle-overview) -->
+
+### Exclusive selection vs. multiple selection
+By default, `md-button-toggle-group` acts like a radio-button group- only one item can be selected.
```
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: radio-button-group (also surround with backticks)

File: src/lib/button-toggle/OVERVIEW.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+In this mode, the `value` of the `md-radio-group` will reflect the value of the selected button and
```
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: md-button-toggle-group?

File: src/lib/button-toggle/OVERVIEW.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+`ngModel` is supported. 
+
+Adding the `multiple` attribute allows multiple items to be selected (checkbox behavior). In this
+mode the values of the the toggles are not used, the `md-radio-group` does not have a value, and
```
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: md-button-toggle-group

File: src/lib/button-toggle/OVERVIEW.md (hunk 4 above)
Comment (jelbourn) [inline src/lib/button-toggle/OVERVIEW.md]: I wanted to refer to the general idea of a radio-button group vs. `md-radio-group` specifically.

**Comment by jelbourn** [comment]
Good catches- comments addressed.

File: src/lib/button-toggle/OVERVIEW.md (hunk 2 above)
Comment (mmalerba) [inline src/lib/button-toggle/OVERVIEW.md]: I still think "buttons toggles" is a weird way to pluralize it, it also says "button toggles" elsewhere in this same readme

**Comment by mmalerba** [comment]
lgtm

**Comment by andrewseguin** [comment]
This had an funny commit in snackbar-ref that is breaking the tests. Check out #2298 for the removal of the line.

**Pull request merged**